src/Host/Client/Impl/rtvs/R/grid.R

```diff
@@ -13,7 +13,28 @@ grid.format <- function(x) {
   sapply(format(x, trim = TRUE, justify = "none"), grid.trim);
 }
 
-grid.data <- function(x, rows, cols) {
+grid.sort.df <- function(x, rows, cols, df.sort.expression) {
+  if(missing(df.sort.expression)) {
+    x.df <- as.data.frame(x)[rows, cols];
+  } else {
+    x.df <- as.data.frame(x)
+    x.df <- x.df[eval(parse(text = df.sort.expression)),][rows, cols];
```

Comment:
Also, side note: while this code already uses dots in identifiers, I would prefer to avoid it in any new code, and stick to either underscores or camelCase. Most R style guides recommend avoiding dots, to avoid clashing with S3 method names.